### ISSUE TITLE
Can't install 3.8.13 nor 3.9.11 as recommended in #2143 resolution

### ISSUE THREAD

**Issue by AhmedKamal** [pyenv/pyenv#2300]
I can not install the following versions as pyenv returns an error although I'm using the latest version of pyenv. 

```
x86 ➜  ~ pyenv install 3.8.13
python-build: definition not found: 3.8.13

See all available versions with `pyenv install --list'.

If the version you need is missing, try upgrading pyenv:

  brew update && brew upgrade pyenv
x86 ➜  ~ pyenv install 3.8.13
python-build: definition not found: 3.8.13

See all available versions with `pyenv install --list'.

If the version you need is missing, try upgrading pyenv:

  brew update && brew upgrade pyenv
```

Diagnostics:

### Description
- Platform information macOS 12.3 
- OS architecture m1 under Rosetta
- pyenv version: 2.2.4, I run brew install pyenv --head
- Python version:
- C Compiler information (e.g. gcc 7.3): 
Apple clang version 13.1.6 (clang-1316.0.21.2)
Target: x86_64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

- Please attach the debug trace of the failing command as a gist:
https://gist.github.com/AhmedKamal/e8e277a46b46f1116316103f50e72281

Closing as per https://github.com/pyenv/pyenv/issues/2143#issuecomment-1070640288:

> I can confirm yesterday's releases 3.7.13, 3.8.13, 3.9.11 and 3.10.3 all work fine on my Intel mac. The GCC solution was not a viable one for me as it caused issues with pip modules that were built with clang. I don't think I need 3.6 or older branches at the moment but I suspect they might need manual patches via pyenv since they're EOL upstream.

**[The cause is a change in Apple Clang 13.0.0+ that made it incompatible with existing Python versions.](https://bugs.python.org/issue45405) Upgrade to the aforementioned newest Python versions to fix the issue.**

---

For older releases, PRs with patches backporting https://github.com/python/cpython/commit/9c4766772cda67648184f8ddba546a5fc0167f91 are welcome if someone is interested in supporting building them with that newer Clang.

_Originally posted by @native-api in https://github.com/pyenv/pyenv/issues/2143#issuecomment-1072032647_

**Comment by native-api** [comment]
These changes are not yet in a release. Install the development version by providing `--head` to `brew install` to get them.

**Comment by AhmedKamal** [comment]
That is exactly what I did

**Comment by native-api** [comment]
On a closer inspection, they are included into `v2.2.5`, but there were some fixes added afterwards still.

We need the diagnostic information as per the issue template to say anything mre specific.

**Comment by AhmedKamal** [comment]
@native-api Updated
